### ISSUE TITLE
Creating team ?

### ISSUE THREAD

**Issue by jeremy-flusin** [mattermost/platform#57]
Hi,

I deployed the docker on a machine on my local network.

When I go to http://dockerhost.my.network.com:8065 (example url), everything looks fine, and I get to the "Enter your team's domain" page.

But then, since I did not have created any team yet, I try to click the "Sign up now" button, which redirects me to http://www.my.network.com:8065/ (according to the example url).

Let me know if you need some additional info.

**Comment by ciuncan** [comment]
+1

**Comment by moomerman** [comment]
I just hit the same issue.  I was accessing it via IP address eg. http://1.2.3.4:8065 and when I click "Sign up now" it redirected to www.222.333.4444:8065.

Managed to work around the issue for now by accessing via http://www.1.2.3.4.xip.io:8065 it seems to need to have the www. at the beginning of the url.  This allowed me to sign up but I haven't received the email yet.

**Comment by poelzi** [comment]
it seems, that without www in the hostname the default docker image does not serve the signup at all. makes it much harder for testing

**Comment by it33** [comment]
Sorry for the late reply--and thank you for getting this far in the installer. Just posted this to another issue: 

If you're running locally as a developer then the system will work with just localhost or dockerhost if mapped in your /etc/hosts file. If you deployed via beanstalk then for now you must have a valid domain that is accessed via www.mydomain.com. Currently you cannot have sub-sub-domains so stuff like www.corp.mydomain.com will not work. It's a known issue we're working on to fix.

**Comment by aaronjlynch** [comment]
Should I be able to create a team yet?  I am not receiving any emails when I sign up.  

Ubuntu 14.04 via docker installation went fine.

**Comment by arkokoley** [comment]
Having the same problem as @aaronjlynch. How do I create a team  without the emails?

**Comment by xyc** [comment]
Have the same issue too

**Comment by ScottBarkman** [comment]
yea, this seems a bit strange - Can't really test if I don't get the confirmation email.

**Comment by cldfzn** [comment]
More than likely your isp is blocking your email attempts or you're being spf filtered.  I set up a mailcatcher docker container (https://quay.io/repository/cldfzn/mailcatcher, still building should be up at some point).
Set up the mailcatcher: docker run -d --name mail -p 1080:1080 quay.io/cldfzn/mailcatcher
Start the mattermost docker container linking mailcatcher: docker run -d --name mattermost -p 8065:80 --link mail:mail mattermost/platform:helium
Then change the config_docker.json EmailSettings to "mail:25" instead of "localhost:25"
 and restart the docker container (or you can modify the Dockerfile and rebuild)

Basically you just need to intercept the mail going out in some manner. I just happened to have a mailcatcher from another project.

**Comment by jschank** [comment]
@it33 
The Docker container implementation instructions indicate that the url for the instance would be dockerhost:8065
However, when I try to set up a team, the URL does not validate because .dockerhost:8065 is more than the limit of 15 characters.
AND there is no way to change that part of the url when setting up the team.

What is the suggested workaround?

**Comment by it33** [comment]
@aaronjlynch @arkokoley @xyc @ScottBarkman 

We've updated the config file to skip email verification. 

You can pull the latest docker container or run the following to skip email verification 
1. `docker exec -ti mattermost-dev /bin/bash` 
2. `cd src/github.com/mattermost/platform/config` 
3. `vi config_docker.json` 
4. `Edit "Mode" : "prod", -> "Mode" : "dev"` 
5. `docker stop mattermost-dev` 
6. `docker start mattermost-dev`

**Comment by aaronjlynch** [comment]
Not to be a pain in the ___, but now nothing seems to work after updating to the latest.

**Comment by dstolpmann** [comment]
Hi,
while setting up a team without e-mail verification works well in the latest Docker image, I can not create a second account using a "Team Invite Link".

It always says after login:
`Please verify your email address. Check your inbox for an email.`

**Comment by cldfzn** [comment]
It's probably still attempting to send emails for everything besides the initial.  Intercepting emails using the method I posted above will get you to the point of using the application at least.

**Comment by dstolpmann** [comment]
OK, it works with MailCatcher. Thank you!

**Comment by larssl** [comment]
The "Sign up now" link has the wrong url in my case. Mattermost is at myhost.mydomain.com:8065 and the link is trying to send me to www.mydomain.com:8065.

It still doesn't work if I use the ip of my remote host, e.g. 192.168.1.1:8065, because the link is now "www.168.1.1:8065"...

**Comment by cldfzn** [comment]
Use an alias in your hosts file "192.168.1.1 www.something.lan" matched to your needs.  Once you get it all set up you'll need to alias "192.168.1.1 teamname.something.lan" as well.

**Comment by sskmani** [comment]
My Config file has the Mode : "dev".
But still it asking for email verification.
How should i prevent this popup.

**Comment by it33** [comment]
Hi everyone, we need your help, 

The Mattermost team wants to see each and every one of your questions answered, either by community or by us. It's really hard to do on GitHub Issues for lack of threading. 

We've setup a forum on http://discourse.mattermost.com, which you can join in one click with your GitHub account. I'm starting to organize some FAQs there, drawing from both the community answers (thank you @cldfzn!) and our internal team.

**Please continue this thread on http://discourse.mattermost.com?**

Apologies again for being so slow at responses, we want community support to be much faster than we're able to offer right now. We just have a lot coming to us across different channels and need just a little more time to get into rhythm.

**Comment by ciuncan** [comment]
@cldfzn Please excuse my ignorance, forgive if I am asking a stupid question, but I am confused about this aliasing, and to be honest don't know much about it. I would very much appreciate a clarification. Should I do it from my laptop, our dev server (on which mattermost is running), or inside the mattermost container? By the way we do not have a domain name attached to our dev server, we only connect with its IP currently.

**Comment by kgcrom** [comment]
@larssl  Temporarily, when you test to mattermost, add host Info below host

```
[your private IP]    beta.mattermo.st         // For Landing page
[your private IP]    www.mattermo.st        // For Sign up page
[your private IP]    [teamname].mattermo.st    // For Team page
```
I speak to one more time, it's temporary. :wink:

**Comment by crspeller** [comment]
#200 Will fix the signup issue. 
#171 Fixed the email issue.

**Comment by dirkk0** [comment]
As a workaround, you can use the console to add a team and team members with:

    ./platform -create_user -team_name="name" -email="user@example.com" -password="mypassword"
    ./platform -create_team -team_name="name" -email="user@example.com"
    ./platform -assign_role -team_name="name" -email="user@example.com" -role="admin"

**Comment by demofly** [comment]
> As a workaround, you can use the console

Excuse me, please. I have no binary called "platform" in my system. how shoud I found it? 
I'm using gitlab-ce_8.2.0-ce.0_amd64.deb for Debian jessie

From /var/log/gitlab/mattermost/mattermost.log :

[2015/11/30 13:18:55 CET] [EROR] /signup/gitlab/complete:AuthorizeOAuthUser code=500 rid=nib41zc6cifaurkmtbsu3wwadh uid= ip=10.8.0.70 Bad response from token request [details: ]

**Comment by demofly** [comment]
> As a workaround, you can use the console

Excuse me, please. I have no binary called "platform" in my system. how shoud I find it? 
I'm using gitlab-ce_8.2.0-ce.0_amd64.deb for Debian jessie

From /var/log/gitlab/mattermost/mattermost.log :
```
[2015/11/30 13:27:21 CET] [EROR] Couldn't find team name=signup, err=We couldn't find the existing team
[2015/11/30 13:32:20 CET] [EROR] /signup/gitlab/complete:AuthorizeOAuthUser code=500 rid=yuxgsu59gpgadx88yre5tkqmiw uid= ip=10.8.0.70 Bad response from token request [details: ]
```

/var/opt/gitlab/mattermost/config.json
```
{
    "ServiceSettings": {
        "ListenAddress": ":8065",
        "MaximumLoginAttempts": 10,
        "SegmentDeveloperKey": "",
        "GoogleDeveloperKey": "",
        "EnableIncomingWebhooks": false,
        "EnablePostUsernameOverride": false,
        "EnablePostIconOverride": false,
        "EnableTesting": false,
        "EnableSecurityFixAlert": true,
        "EnableOutgoingWebhooks": false
    },
    "TeamSettings": {
        "SiteName": "Team",
        "MaxUsersPerTeam": 150,
        "EnableTeamCreation": true,
        "EnableUserCreation": true,
        "RestrictCreationToDomains": "",
        "RestrictTeamNames": true,
        "EnableTeamListing": false
    },
    "SqlSettings": {
        "DriverName": "postgres",
        "DataSource": "user=gitlab_mattermost host=/var/opt/gitlab/postgresql port=5432 dbname=mattermost_production",
        "DataSourceReplicas": ["user=gitlab_mattermost host=/var/opt/gitlab/postgresql port=5432 dbname=mattermost_production"],
        "MaxIdleConns": 10,
        "MaxOpenConns": 10,
        "Trace": false,
        "AtRestEncryptKey": "xxx"
    },
    "LogSettings": {
        "ConsoleEnable": true,
        "ConsoleLevel": "INFO",
        "EnableFile": true,
        "FileLevel": "ERROR",
        "FileFormat": "",
        "FileLocation": "/var/log/gitlab/mattermost/mattermost.log"
    },
    "FileSettings": {
        "DriverName": "local",
        "Directory": "/var/opt/gitlab/mattermost/data",
        "EnablePublicLink": true,
        "PublicLinkSalt": "xxx",
        "ThumbnailWidth": 120,
        "ThumbnailHeight": 100,
        "PreviewWidth": 1024,
        "PreviewHeight": 0,
        "ProfileWidth": 128,
        "ProfileHeight": 128,
        "InitialFont": "luximbi.ttf",
        "AmazonS3AccessKeyId": "",
        "AmazonS3SecretAccessKey": "",
        "AmazonS3Bucket": "",
        "AmazonS3Region": ""
    },
    "EmailSettings": {
        "EnableSignUpWithEmail" : false,
        "SendEmailNotifications": false,
        "RequireEmailVerification": false,
        "SMTPUsername": "",
        "SMTPPassword": "",
        "SMTPServer": "",
        "SMTPPort": "",
        "ConnectionSecurity": "",
        "InviteSalt": "xxx",
        "PasswordResetSalt": "xxx",
        "FeedbackName": "",
        "FeedbackEmail": "",
        "ApplePushServer": "",
        "ApplePushCertPublic": "",
        "ApplePushCertPrivate": ""
    },
    "RateLimitSettings": {
        "EnableRateLimiter": true,
        "PerSec": 10,
        "MemoryStoreSize": 10000,
        "VaryByRemoteAddr": true,
        "VaryByHeader": ""
    },
    "PrivacySettings": {
        "ShowEmailAddress": true,
        "ShowFullName": true
    },
    "GitLabSettings": {
        "Enable": true,
        "Secret": "xxx",
        "Id": "xxx",
        "Scope": "",
        "AuthEndpoint": "http://gitlab.mycompany.com/oauth/authorize",
        "TokenEndpoint": "http://gitlab.mycompany.com/oauth/token",
        "UserApiEndpoint": "http://gitlab.mycompany.com/api/v3/user"
    }
}

```

**Comment by dirkk0** [comment]
I don't have a running installation right now, so I can't tell you atm. But I followed the (non Docker) installation, and it happened to be there.

**Comment by demofly** [comment]
Hi there! Unexpectedly, Mattermost silently fails until I've set SMTP parameters. Dunno why it had any sense while I'm using Gitlab SSO mode without e-mailing.
